templates/consul.systemd.epp

```diff
@@ -0,0 +1,37 @@
+<%- | Hash $config_hash,
+      String[1] $user,
+      String[1] $group,
+      Stdlib::Absolutepath $bin_dir,
+      Stdlib::Absolutepath $config_dir,
+      Optional[String] $extra_options,
+      Boolean $allow_binding_to_root_ports,
+      Boolean $enable_beta_ui
+| -%>
+# THIS FILE IS MANAGED BY PUPPET
+[Unit]
+Description=Consul Agent
+After=network-online.target
+Wants=network-online.target
+
+[Service]
+<% if $config_hash['retry_join'] == undef or $config_hash['retry_join'].length < 2 { -%>
```

Comment:
I checked https://github.com/voxpupuli/puppet-consul/pull/552 and I'm not sure if this is really required. But we had the logic in the erb template, so I just ported it.

Reply:
If you're changing it, I'd keep the `config_hash` of the template and move it into the Puppet code. Introduce a simple `String[1] $service_type` (or `Enum`) parameter. Mostly since eventually I'd consider using [`systemd::manage_unit`](https://github.com/voxpupuli/puppet-systemd/blob/master/manifests/manage_unit.pp) once that's out in a release.